src/pal/src/misc/sysinfo.cpp

```diff
@@ -343,5 +343,8 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     cacheSize = max(cacheSize, sysconf(_SC_LEVEL4_CACHE_SIZE));
 #endif
 
+    if(cacheSize == 0)
+      cacheSize = 8ULL << 20;  // sysconf currently only reports cache size info for x86.
```

Comment:
Have you actually hit a problem with the current behavior? It is only used by the GC for estimating the gen0 segment size and the code works ok with the cache size reported as 0, setting the lower cap of the gen0 at 256 kB (otherwise it would be cache size divided by the number of cores).
The 8MB would be an overestimation e.g. for ARM Cortex A9, where the L2 cache is 1MB. Overestimating it may have negative impact on the GC performance. 
Is there really no way to get the cache size on Linux, e.g. reading the proc file system or some other way? If that's the case, I would suggest making this value at least 32 / 64 bit specific.
CC: @Maoni0 what do you think?